### PR TITLE
upgrade sql sqlalchemy

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description
upgrade sql sqlalchemy

## How Has This Been Tested?

CI tests run

- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by roshnaeem** [comment]
@peterdudfield, so I checked by reverting the changes, and checked the tests without any changes, and on an old version of sqlalchemy = 1.4.46, it failed one test, which I\is the same error which I tried to fix after updating the version. Because of this, I replaced List with Mapped, to fix that error. 
However, it seems as if this test case was already failing, there might be some other problem and is not related to updating the version of sqlalchemy. Maybe we can just consider this PR #107. I can't see the tests which #107 was failing. Can you help me with that.

**Comment by peterdudfield** [comment]
> @peterdudfield, so I checked by reverting the changes, and checked the tests without any changes, and on an old version of sqlalchemy = 1.4.46, it failed one test, which I\is the same error which I tried to fix after updating the version. Because of this, I replaced List with Mapped, to fix that error. I tested it with pulling the code from main branch too.
> 
> Screenshot 2024-03-27 at 1 46 56 AM
> However, it seems as if this test case was already failing, there might be some other problem and is not related to updating the version of sqlalchemy. Maybe we can just consider this PR #107. I can't see the tests which #107 was failing. Can you help me with that.

So the tests on `main` right now are passing. So I think this failure is due to the upgrade here. Maybe you need to pull the latest branch from main into your branch

File: pvsite_datamodel/sqlmodels.py [hunk 1]
```diff
@@ -155,12 +154,12 @@ class SiteSQL(Base, CreatedMixin):
...
-    forecasts: List["ForecastSQL"] = relationship("ForecastSQL", back_populates="site")
-    generation: List["GenerationSQL"] = relationship("GenerationSQL")
-    inverters: List["InverterSQL"] = relationship(
+    forecasts: Mapped["ForecastSQL"] = relationship("ForecastSQL", back_populates="site")
+    generation: Mapped["GenerationSQL"] = relationship("GenerationSQL")
+    inverters: Mapped["InverterSQL"] = relationship(
         "InverterSQL", back_populates="site", cascade="all, delete-orphan"
     )
-    site_groups: List["SiteGroupSQL"] = relationship(
+    site_groups: Mapped["SiteGroupSQL"] = relationship(
```
Comment (peterdudfield) [inline pvsite_datamodel/sqlmodels.py:162]: I wonder if this needs to be `Mapped[List["SiteGroupSQL"]]` or Mapped[["SiteGroupSQL"]], or something like that

**Comment by peterdudfield** [comment]
Thanks @roshnaeem for all your work on this

**Pull request merged**